### DISCUSSION TITLE
Is it possible to connect to a database on S3?

### DISCUSSION THREAD

**Discussion by dhirschfeld** [duckdb/duckdb#10466] [Q&A] [answered]
When I try to connect to an `s3` url it appends it to my local `/home/dhirschfeld` path, even if I have registered the s3 filesystem:
```python
>>> import duckdb
>>> import fsspec
>>> bucket = '<bucket>'
>>> duckdb.list_filesystems()
[]

>>> fs = fsspec.filesystem('s3')
>>> duckdb.register_filesystem(fs)
>>> duckdb.list_filesystems()
['s3']

>>> duckdb.connect(f"s3://{bucket}/test.db")
---------------------------------------------------------------------------
IOException                               Traceback (most recent call last)
Cell In[10], line 1
----> 1 duckdb.connect(f"s3://{bucket}/test.db")

IOException: IO Error: Cannot open file "/home/dhirschfeld/s3://<bucket>/test.db": No such file or directory
```

**Comment by carlopi** [comment, marked as answer]
This issue looks somehow related: https://github.com/duckdb/duckdb/issues/10409, probably conversation is better moved there.

Can you try if you can access the publicly accessible DB file at:
```
ATTACH 's3://duckdb-blobs/data/my.db' as test (READ_ONLY);
```
?

**Comment by dhirschfeld** [comment]
I also just found:
 - https://github.com/duckdb/duckdb/discussions/10029
 - https://github.com/duckdb/duckdb/discussions/8893#discussioncomment-6978758
 
 ...and in the latter is was confirmed that only read is supported. Since I need to be able to write to the database that won't help in my case - it seems what I want (read/write access to a database on s3) isn't possible :(
